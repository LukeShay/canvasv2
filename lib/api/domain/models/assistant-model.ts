/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, RelationMappings, JSONSchema } from 'objection';
import Constants from '../constants';

export class AssistantModel extends Model {
  static tableName = Constants.ASSISTANTS_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        assistantId: { type: 'string' },
        classId: { type: 'string' },
      },
      required: ['assistantId', 'classId'],
      type: 'object',
    };
  }

  static get relationMappings(): RelationMappings {
    // ts-expect-error
    const { UserModel } = require('./user-model');
    // ts-expect-error
    const { ClassModel } = require('./class-model');

    return {
      assistant: {
        join: {
          from: `${AssistantModel.tableName}.assistantId`,
          to: `${UserModel.tableName}.${UserModel.idColumn}`,
        },
        modelClass: UserModel,
        relation: Model.BelongsToOneRelation,
      },
      class: {
        join: {
          from: `${AssistantModel.tableName}.classId`,
          to: `${ClassModel.tableName}.${ClassModel.idColumn}`,
        },
        modelClass: ClassModel,
        relation: Model.BelongsToOneRelation,
      },
    };
  }
}
