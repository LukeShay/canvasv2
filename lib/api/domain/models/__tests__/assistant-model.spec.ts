import { AssistantModel } from '../assistant-model';

describe('AssistantModel', () => {
  it("should have idColum 'id'", () => {
    expect(AssistantModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(AssistantModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(AssistantModel.relationMappings).toMatchSnapshot();
  });
});
