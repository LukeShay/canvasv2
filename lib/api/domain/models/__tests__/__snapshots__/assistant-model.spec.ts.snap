// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssistantModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "assistantId": Object {
      "type": "string",
    },
    "classId": Object {
      "type": "string",
    },
  },
  "required": Array [
    "assistantId",
    "classId",
  ],
  "type": "object",
}
`;

exports[`AssistantModel should should have relationMappings that matches snapshot 1`] = `
Object {
  "assistant": Object {
    "join": Object {
      "from": "Assistants.assistantId",
      "to": "Users.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
  "class": Object {
    "join": Object {
      "from": "Assistants.classId",
      "to": "Classes.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
